source/fundamentals/promises.txt

```diff
@@ -0,0 +1,194 @@
+======================
+Promises and Callbacks
+======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecols
+
+Overview
+--------
+
+The NodeJS driver uses the asynchronous Javascript API to communicate with
+your MongoDB cluster.
+
+Asynchronous Javascript allows you to execute operations without waiting for
+the processing thread to become free. This helps prevent your application
+from becoming unresponsive to web UI rendering or network requests when
+executing long-running operations. For more information about asynchronous
```

Comment:
```suggestion
from becoming unresponsive when
executing long-running operations. For more information about asynchronous
```

Reply:
I noticed I repeated nearly the same sentence, so I'm getting rid of one.